velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -237,8 +237,21 @@ SubstraitVeloxExprConverter::toVeloxExpr(
     const RowTypePtr& inputType) {
   auto substraitType = subParser_->parseType(castExpr.type());
   auto type = toVeloxType(substraitType->type);
-  // TODO add flag in substrait after. now is set false.
-  bool nullOnFailure = false;
+  auto failureBehavior =  castExpr.failure_behavior();
+  bool nullOnFailure;
+  switch (failureBehavior) {
+    case ::substrait::Expression_Cast_FailureBehavior_FAILURE_BEHAVIOR_UNSPECIFIED:
+      nullOnFailure = false;
```

Comment:
What is the default behavior of Spark? Maybe we can keep aligned with that.

Reply:
By default, spark behaves like `nullOnFailure = true`. We have made some changes in gluten code to tell velox which behavior should be used according to spark's config through substrait.

Reply:
A supplement:
In velox, the default behavior is like `nullOnFailure = false`. So for unspecified case, we keep using such setting.